src/Core/Compiler/TemplateCompiler.php

```diff
@@ -37,13 +39,49 @@ class TemplateCompiler {
 	protected $renderingContext;
 
 	/**
+	 * @var string
+	 */
+	protected $mode = self::MODE_NORMAL;
```

Comment:
ok, maybe i'm confusing this with a ``new ClassName()`` something wasn't possible there well beyond 5.0, afaik